packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -479,6 +480,12 @@ export interface SdkPathParameter extends SdkModelPropertyTypeBase {
   correspondingMethodParams: SdkModelPropertyType[];
 }
 
+export interface SdkCookieParameter extends SdkModelPropertyTypeBase {
```

Comment:
i don't think tcgc should swallow any parameter. emitter could choice whether ignore it.

Reply:
this might cause issues in consuming emitters, just because since we're adding another type here, so previous switch statements / if else checks might be affected. I'm ok with this though, we should just make sure people are aware.

Reply:
We need to figure out what is acceptable in the future but I think a crash because you don't handle a new type is ok as long as this new type doesn't show up on existing specs. If the TypeScript build fails because of it its also a different problem from the runtime failing.
In general I think the following needs to be true when working with ts unions for us to keep adding new features:
1. adding a new type to a union shouldn't count as breaking anyone as long as:
   a. the type only show up for scenarios that weren't possible before
2. if an library TS compilation fails uptaking the new version because they don't handle the new type its a non issue, it doesn't affect costumers.
3. if the library crash because of the new type(As long as `1.a` is respected) , this is not ideal but this is also an acceptable temporary state of the library until that new type is either handled or explicitly made an error/warning and dropped.

Reply:
agree. i'll let all language emitter's dev to confirm this change.